### ISSUE TITLE
[Performance] CTCLabelDecode redundantly copies NumPy logits returned by inference runners

### ISSUE THREAD

**Issue by 1443858742** [PaddlePaddle/PaddleX#5181]
## Checklist

- [x] Searched existing open and closed issues and pull requests
- [x] Checked the PaddleX FAQ and deployment documentation
- [x] Confirmed the behavior still exists in both `release/3.7` and `develop`

## Problem

`CTCLabelDecode.__call__` currently performs an unconditional copy:

```python
preds = np.array(pred[0])
preds_idx = preds.argmax(axis=-1)
preds_prob = preds.max(axis=-1)
```

For the current `paddle_static` runner, `pred[0]` is already a NumPy array returned by `copy_to_cpu()`. The additional `np.array()` therefore copies the complete recognition logits a second time before the read-only `argmax` and `max` operations.

This becomes expensive for models with a large CTC vocabulary such as PP-OCRv6 Medium.

Relevant sources:

- CTC postprocessor: https://github.com/PaddlePaddle/PaddleX/blob/develop/paddlex/inference/models/text_recognition/processors.py
- Paddle static runner: https://github.com/PaddlePaddle/PaddleX/blob/develop/paddlex/inference/models/runners/paddle_static/runner.py
- HPI runner: https://github.com/PaddlePaddle/PaddleX/blob/develop/paddlex/inference/models/runners/hpi/runner.py

## Reproduction and measurements

Environment:

- PaddleX 3.7.2
- PaddleOCR 3.7.0
- PaddlePaddle GPU 3.3.1, CUDA 12.6
- Python 3.11
- PP-OCRv6 Medium, FP32, `paddle_static`
- NVIDIA RTX 3090

A complex page produced 113 OCR lines in 19 recognition batches. The model output has 18,710 CTC classes and the page produced 794.45 MiB of logits in total.

An isolated A/B test changed only:

```diff
- preds = np.array(pred[0])
+ preds = np.asarray(pred[0])
```

Results:

| Test | Official | No-copy | Result |
| --- | ---: | ---: | ---: |
| Local NUMA, 41 stratified pages, mean | 1.156 s | 0.848 s | 25.47% mean paired reduction |
| Local NUMA, p95 | 1.838 s | 1.385 s | lower tail latency |
| Cross-NUMA, 34 paired observations, mean | 3.200 s | 1.522 s | 31.26% mean paired reduction |
| Cross-NUMA, p95 | 10.897 s | 2.429 s | large tail reduction |

The direct CTC array-conversion time changed from:

- Local NUMA: 0.2472 s mean to about 21 microseconds
- Cross-NUMA: 1.2089 s mean to about 22 microseconds

The cross-NUMA figures include a separate process-affinity problem; the redundant copy is not the only NUMA issue, but it strongly amplifies the memory traffic and tail latency.

## Output validation

Across 75 paired observations:

- OCR line count: exact match
- Character count: exact match
- Title text hash: exact match
- Full text hash: exact match
- Parsed OCR-line JSON hash: exact match
- Semantic-field JSON hash: exact match

There were 0 output mismatches.

## Compatibility consideration

The conversion may have originally normalized outputs from different runners. Removing conversion entirely may break callers that pass a Paddle tensor, Python list, or another array-like object.

However, current PaddleX runner interfaces return `List[np.ndarray]` for the standard Paddle and HPI paths. PaddleOCR's current CTC decoder also handles tensors explicitly and uses existing NumPy arrays without forcing another copy:

https://github.com/PaddlePaddle/PaddleOCR/blob/main/ppocr/postprocess/rec_postprocess.py

A compatibility-preserving implementation could keep existing NumPy arrays and only convert other array-like values. The proposed PR will include compatibility tests and verify that NumPy input shares memory rather than being copied.

## Expected behavior

CTC decoding should preserve support for existing runner output types while avoiding a second full copy when the runner has already returned a NumPy array.
